host/xtest/sdp_basic.c

```diff
@@ -570,9 +570,9 @@ static int invoke_out_of_bounds(struct tee_ctx *ctx,
 	 * result is.
 	 */
 	if ((valid_ref && orig != TEEC_ORIGIN_TRUSTED_APP) ||
-	    (!valid_ref && orig != TEEC_ORIGIN_COMMS &&
-	     (teerc != TEEC_ERROR_GENERIC ||
-	      teerc != TEEC_ERROR_BAD_PARAMETERS)))
```

Comment:
Should test orig != TEEC_ORIGIN_TRUSTED_APP:
```c
	if ((valid_ref && orig != TEEC_ORIGIN_TRUSTED_APP) ||
	    (!valid_ref && (orig == TEEC_ORIGIN_TRUSTED_APP ||
			    (teerc != TEEC_ERROR_GENERIC &&
			     teerc != TEEC_ERROR_BAD_PARAMETERS)))
		goto error;
```

That makes a bit complex case. I would suggest to make it more readable in 2 steps. The very same, yes, but a bit easier to read out, I think.
```c
	if (valid_ref && orig != TEEC_ORIGIN_TRUSTED_APP)
		goto error;

	if (!valid_ref && (orig == TEEC_ORIGIN_TRUSTED_APP ||
			   (teerc != TEEC_ERROR_GENERIC &&
			    teerc != TEEC_ERROR_BAD_PARAMETERS)))
		goto error;

Reply:
sorry, I should have read the whole P-R...